src/helpers/optionalStacking/common.ts

```diff
@@ -28,24 +37,111 @@ export function getViewByTwoAttributes(
  *      => ignore computeRatio and use stackMeasuresToPercent
  *
  * stackMeasures is applied only when there are [2 measures and more]
- * stackMeasuresToPercent is applied only when there are [1 measure + 1 stackBy] or [2 measures and up]
+ * stackMeasuresToPercent is applied only when there are [1 measure + 1 stackBy] or [2 measures and more]
  */
 export function getSanitizedStackingConfig(
+    buckets: VisualizationObject.IBucket[],
+    config: IChartConfig = {},
+): ISanitizedStackingConfig {
+    const clonedBuckets = cloneDeep(buckets);
+    const measuresItems = findBucketByLocalIdentifier(clonedBuckets, MEASURES).items || [];
+
+    if (measuresItems.length > 1) {
+        return {
+            config,
+            buckets: disableComputeRatio(clonedBuckets),
+        };
+    }
+
+    if (measuresItems.length === 1) {
+        const isComputeRatio = getComputeRatio(measuresItems[0]);
+        const hasStackByAttribute = isStackedChart(clonedBuckets);
+        const newBuckets =
+            isComputeRatio && hasStackByAttribute ? ignoreStackBucket(clonedBuckets) : clonedBuckets;
+        const { stackMeasures, stackMeasuresToPercent } = config;
+
+        return {
+            buckets: newBuckets,
+            config: {
+                ...config,
+                stackMeasures: stackMeasures && !isComputeRatio,
+                stackMeasuresToPercent: stackMeasuresToPercent && !isComputeRatio && hasStackByAttribute,
+            },
+        };
+    }
+
+    return {
+        buckets,
+        config,
+    };
+}
+
+function disableComputeRatio(buckets: VisualizationObject.IBucket[]): VisualizationObject.IBucket[] {
+    return buckets.map((bucket: VisualizationObject.IBucket) => {
+        if (bucket.localIdentifier === MEASURES) {
```

Comment:
can be replaced by findBucketByLocalIdentifier

Reply:
`disableComputeRatio` returns the whole buckets, not only the `MEASURES` bucket.
So I can not use `findBucketByLocalIdentifier` here.

Reply:
ok